iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/setup/DeviceTwinCommon.java

```diff
@@ -431,6 +431,17 @@ public DeviceTwinTestInstance(IotHubClientProtocol protocol, AuthenticationType
             this.privateKey = privateKey;
             this.x509Thumbprint = x509Thumbprint;
             this.clientType = clientType;
+
+            try
+            {
+                this.sCDeviceTwin = DeviceTwin.createFromConnectionString(iotHubConnectionString);
+            }
+            catch (IOException e)
+            {
+                // A little odd to do this, but the createFromConnectionString call doesn't ever actually throw this exception, and
+                // this constructor shouldn't throw
+                throw new RuntimeException("Unexpected exception encountered while building device twin service client", e);
```

Comment:
I did look into this IOException and verified that it isn't actually ever thrown. I'd remove the exception from the method signature, but that would be a breaking change